### ISSUE TITLE
Capturing errors from package manager

### ISSUE THREAD

**Issue by piersc** [rackerlabs/auter#27]
During prep, last-prep-config is created containing only stdout from the yum/dnf run. stderr is not captured - it'll currently get printed to screen when running on a tty, or use cron's behaviour, which by default is to send an email. This is the kind of message that could be on stderr, but won't prevent the pkgmanager command from succeeding:

```
There are unfinished transactions remaining. You might consider running yum-complete-transaction, or "yum-complete-transaction --cleanup-only" and "yum history redo last", first to finish them. If those don't work you'll have to try removing/installing packages by hand (maybe package-cleanup can help).
```

Apply behaves differently - all output is discarded from the update command, but the yum history command output is saved, which includes any errors.

**Comment by piersc** [comment]
It appears when running apply some errors are actually sent to stderr. It seems these are coming from https://github.com/rackerlabs/auter/blob/master/auter#L163

```
 local HISTORY_BEFORE=$(${PACKAGE_MANAGER} history list)
```

Here's an example of what this might show:

```
# auter --apply
INFO: Running with: /usr/bin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Applying updates
Warning: RPMDB altered outside of yum.
** Found 10 pre-existing rpmdb problem(s), 'yum check' output follows:
avahi-autoipd-0.6.31-15.el7_2.1.x86_64 is a duplicate with avahi-autoipd-0.6.31-15.el7.x86_64
avahi-libs-0.6.31-15.el7_2.1.x86_64 is a duplicate with avahi-libs-0.6.31-15.el7.x86_64
ca-certificates-2015.2.6-70.1.el7_2.noarch is a duplicate with ca-certificates-2015.2.4-71.el7.noarch
coreutils-8.22-15.el7_2.1.x86_64 is a duplicate with coreutils-8.22-15.el7.x86_64
cyrus-sasl-lib-2.1.26-20.el7_2.x86_64 is a duplicate with cyrus-sasl-lib-2.1.26-19.2.el7.x86_64
cyrus-sasl-plain-2.1.26-20.el7_2.x86_64 is a duplicate with cyrus-sasl-plain-2.1.26-19.2.el7.x86_64
1:grub2-tools-2.02-0.34.el7.centos.x86_64 is a duplicate with 1:grub2-tools-2.02-0.29.el7.centos.x86_64
python-pyudev-0.15-7.el7_2.1.noarch is a duplicate with python-pyudev-0.15-7.el7.noarch
selinux-policy-3.13.1-60.el7_2.3.noarch is a duplicate with selinux-policy-3.13.1-60.el7.noarch
selinux-policy-targeted-3.13.1-60.el7_2.3.noarch is a duplicate with selinux-policy-targeted-3.13.1-60.el7.noarch
INFO: Updates complete (yum Transaction ID : 57). You may need to reboot for some updates to take effect
```

**Comment by piersc** [comment]
To summarise this:
1. During prep, stderr is not captured to file, but is printed to screen
2. During apply, errors from the yum history command will do the same
3. During apply, no errors are either captured or printed to screen, the output of yum is dropped

Point 3 makes debugging what went wrong impossible without rerunning yum manually. 1 and 2 means the error output won't be in the place you expect it to be.

**Comment by reaperzn** [comment]
fixed in #36
